subsys/bluetooth/controller/ll_sw/ull_iso.c

```diff
@@ -243,47 +258,43 @@ uint8_t ll_setup_iso_path(uint16_t handle, uint8_t path_dir, uint8_t path_id,
 			  uint16_t vs_codec_id, uint32_t controller_delay,
 			  uint8_t codec_config_len, uint8_t *codec_config)
 {
-	ARG_UNUSED(controller_delay);
-	ARG_UNUSED(codec_config_len);
-	ARG_UNUSED(codec_config);
-
-	if (path_id == BT_HCI_DATAPATH_ID_DISABLED) {
-		return 0;
-	}
-
-#if defined(CONFIG_BT_CTLR_SYNC_ISO)
-	if (path_dir != BT_HCI_DATAPATH_DIR_CTLR_TO_HOST) {
-		/* FIXME: workaround to succeed datapath setup for ISO
-		 *        broadcaster until Tx datapath is implemented, in the
-		 *        future.
-		 */
-		return BT_HCI_ERR_SUCCESS;
-	}
-#endif /* CONFIG_BT_CTLR_SYNC_ISO */
+	struct ll_iso_datapath *dp;
 
 #if defined(CONFIG_BT_CTLR_SYNC_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
-	isoal_sink_handle_t sink_handle;
+	struct lll_sync_iso_stream *sync_stream = NULL;
+	struct ll_conn_iso_stream *cis = NULL;
 	uint32_t stream_sync_delay;
 	uint32_t group_sync_delay;
 	uint8_t flush_timeout;
 	uint16_t iso_interval;
 	uint32_t sdu_interval;
 	uint8_t  burst_number;
-	isoal_status_t err;
+	uint8_t max_octets;
 	uint8_t framed;
 	uint8_t role;
+#endif /* CONFIG_BT_CTLR_SYNC_ISO || CONFIG_BT_CTLR_CONN_ISO */
 
-#if defined(CONFIG_BT_CTLR_CONN_ISO)
-	struct ll_conn_iso_stream *cis = NULL;
-	struct ll_conn_iso_group *cig = NULL;
-	isoal_source_handle_t source_handle;
-	uint8_t max_octets;
+	ARG_UNUSED(controller_delay);
+	ARG_UNUSED(codec_config);
 
-	if (IS_CIS_HANDLE(handle)) {
+	if (path_id == BT_HCI_DATAPATH_ID_DISABLED) {
+		return BT_HCI_ERR_SUCCESS;
+	}
+
+	if (false) {
+
+#if defined(CONFIG_BT_CTLR_CONN_ISO)
+	} else if (IS_CIS_HANDLE(handle)) {
+		struct ll_conn_iso_group *cig;
 		struct ll_conn *conn;
 
+		/* If the Host attempts to set a data path with a Connection
+		 * Handle that does not exist or that is not for a CIS or a BIS,
+		 * the Controller shall return the error code Unknown Connection
+		 * Identifier (0x02)
+		 */
 		cis = ll_conn_iso_stream_get(handle);
-		if (!cis->group) {
+		if (!cis || !cis->group) {
```

Comment:
`cis` cannot be NULL and should not be checked here (usual design pattern)

Reply:
I don't see the update?

Reply:
hmmm... a rebase of some commits remove it. Fixed again.